src/Symfony/Component/Mailer/Transport/Smtp/EsmtpTransport.php

```diff
@@ -192,7 +193,6 @@ private function handleAuth(array $modes): void
                 continue;
             }
 
-            $code = null;
```

Comment:
shouldn't we still reset it on each iteration of the loop ?

Reply:
yes, you are right. I added it in my forked branch should I create a new PR?

Reply:
Please do :)

Reply:
Here it is: #51581 :)